src/main/java/com/liorkn/elasticsearch/plugin/VectorScoringPlugin.java

```diff
@@ -13,23 +13,167 @@
+                    }
+
+                    @Override
+                    public SearchScript newInstance(LeafReaderContext context) throws IOException {
+                        return new SearchScript(p, lookup, context) {
+                        	BinaryDocValues docAccess = context.reader().getBinaryDocValues(field);
+                            int currentDocid = -1;
+                            
+                            @Override
+                            public void setDocument(int docid) {
+                                // Move to desired document
+                                try {
+                                	docAccess.advanceExact(docid);
+                                } catch (IOException e) {
+                                    throw new UncheckedIOException(e);
+                                }
+                                currentDocid = docid;
+                            }
+                            
+                            @Override
+                            public double runAsDouble() {
+                            	if (currentDocid < 0) {
+                            		return 0.0;
+                            	}
+                            	//actually run scoring
+                            	final int size = searchVector.size();
+
+                            	try {
+                                    final byte[] bytes = docAccess.binaryValue().bytes;
+                                    final ByteArrayDataInput input = new ByteArrayDataInput(bytes);
+                                    input.readVInt(); // returns the number of values which should be 1, MUST appear hear since it affect the next calls
+                                    final int len = input.readVInt(); // returns the number of bytes to read//if submitted vector is different size
+                                    if (len != size * DOUBLE_SIZE) {
+                                        return 0.0;
+                                    }
+                                    
+                                    final int position = input.getPosition();
+                                    final DoubleBuffer doubleBuffer = ByteBuffer.wrap(bytes, position, len).asDoubleBuffer();
 
+                                    final double[] docVector = new double[size];
+                                    doubleBuffer.get(docVector);
+                                    double docVectorNorm = 0.0f;
+                                    double score = 0;
+                                    for (int i = 0; i < size; i++) {
+                                        // doc inputVector norm
+                                        if(cosine) {
+                                            docVectorNorm += docVector[i]*docVector[i];
+                                        }
+                                        // dot product
+                                        score += docVector[i] * searchVector.get(i).doubleValue();
```

Comment:
Since this line is done <num of vectors in corpus>*<vector dimension> times, it needs to be as fast as possible.
For performance reasons searchVector should be a primitive type